### PR TITLE
Replace fog with aws sdk

### PR DESCRIPTION
**Pull request by nahiluhmot**
@tlunter

### CODE REVIEW

File: dockly.gemspec [hunk 1]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency 'docker-api', '>= 1.14.0'
   gem.add_dependency 'dockly-util', '~> 0.0.9'
   gem.add_dependency 'excon'
-  gem.add_dependency 'fog', '~> 1.21'
+  gem.add_dependency 'aws-sdk', '~> 2.1'
```
Comment (tlunter) [inline dockly.gemspec]: Any reason to restrict to 2.1+ only? shouldn't 2.0 be fine?

Comment (nahiluhmot) [inline dockly.gemspec]: Good call.

File: lib/dockly/build_cache/base.rb [hunk 2]
```diff
@@ -48,9 +48,9 @@ def insert_latest
 
   def up_to_date?
     ensure_present! :s3_bucket, :s3_object_prefix
-    connection.head_object(s3_bucket, s3_object(hash_output))
+    connection.head_object(bucket: s3_bucket, key: s3_object(hash_output))
     true
-  rescue Excon::Errors::NotFound
+  rescue Aws::S3::Errors::ServiceError
```
Comment (tlunter) [inline lib/dockly/build_cache/base.rb]: What about using the NoSuchKey exception and not just ServiceError?

File: lib/dockly/build_cache/base.rb [hunk 3]
```diff
@@ -63,17 +63,17 @@ def pull_from_s3(version)
...
       file = File.open(file_path, 'w+b')
-      file.write(object.body)
+      file.write(object.body.read)
       file.tap(&:rewind)
     else
       info 'Build cache already exists locally'
       File.open(file_path, 'rb')
     end
-  rescue Excon::Errors::NotFound
+  rescue Aws::S3::Errors::ServiceError
```
Comment (tlunter) [inline lib/dockly/build_cache/base.rb]: Same with this one.

File: lib/dockly/deb.rb [hunk 4]
```diff
@@ -63,7 +67,7 @@ def build_path
 
   def exists?
     debug "#{name}: checking for package: #{s3_url}"
-    Dockly::AWS.s3.head_object(s3_bucket, s3_object_name)
+    Dockly.s3.head_object(bucket: s3_bucket, key: s3_object_name)
     info "#{name}: found package: #{s3_url}"
     true
   rescue
```
Comment (tlunter) [inline lib/dockly/deb.rb]: We should probably limit this rescue to the NoSuchKey exception.

**Comment by tlunter** [comment]
:+1: Good to go pending successful builds.

**Pull request merged**